File: docs/examples/js/src/data_management_update_labels.mjs
import {Client} from "reduct-js";
import assert from "node:assert";

// Create a client instance, then get or create a bucket
const client = new Client("http://127.0.0.1:8383", {apiToken: "my-token"});
const bucket = await client.getOrCreateBucket("bucket");

// Send some records to the "entry" entry with labels
const timestamp = BigInt(Date.now()) * 1000n;
let record = await bucket.beginWrite("entry-1", {
    ts: timestamp,
    labels: {"key1": "value1", "key2": "value2"},
});
await record.write("Some binary data");

record = await bucket.beginWrite("entry-1", {
    ts: timestamp + 1000_000n,
    labels: {"key1": "value1", "key2": "value2"},
});
await record.write("Some more binary data");

// Update labels of a record: remove "key2" and update "key1"
await bucket.update("entry-1", timestamp, {"key1": "new-value", "key2": ""});
record = await bucket.beginRead("entry-1", timestamp, true);    // only labels
assert(record.labels["key1"] === "new-value");
assert(record.labels["key2"] === undefined);


// Update labels in a batch
const batch = await bucket.beginUpdateBatch("entry-1");
batch.addOnlyLabels(timestamp, {label1: "value1", label2: ""});
batch.addOnlyLabels(timestamp + 1000_000n, {label3: "value3"});
const errors = await batch.write();
assert(errors.size === 0);

File: versioned_docs/version-1.20.x/examples/go/src/data_management_update_labels.go
package main

import (
	"context"
	reduct "github.com/reductstore/reduct-go"
	"time"
)

func main() {
	// Create a client and use the base URL and API token
	client := reduct.NewClient("http://localhost:8383", reduct.ClientOptions{
		APIToken: "my-token",
	})

	// Get or create a bucket with the name "my-bucket"
	bucket, err := client.CreateOrGetBucket(context.Background(), "my-bucket", nil)
	if err != nil {
		panic(err)
	}

	// Send some records to the "go-example"
	ts := time.Now().UnixMicro()
	err = bucket.BeginWrite(context.Background(), "go-example", &reduct.WriteOptions{
		Timestamp: ts,
	}).Write([]byte("Some binary data"))
	if err != nil {
		panic(err)
	}

	err = bucket.BeginWrite(context.Background(), "go-example", &reduct.WriteOptions{
		Timestamp: ts + 1,
	}).Write([]byte("Some binary data"))
	if err != nil {
		panic(err)
	}

	// Update labels of a record: remove "key2" and update "key1"
	err = bucket.Update(context.Background(), "go-example", ts, map[string]any{"key1": "new-value", "key2": ""})
	if err != nil {
		panic(err)
	}

	record, err := bucket.BeginRead(context.Background(), "go-example", &ts)
	if err != nil {
		panic(err)
	}

	if record.Labels()["key1"] != "new-value" || record.Labels()["key2"] != nil {
		panic("Labels not updated correctly")
	}

	// Update labels in a batch
	batch := bucket.BeginUpdateBatch(context.Background(), "go-example")
	batch.AddOnlyLabels(ts, map[string]any{"key1": "new-value", "key2": ""})
	batch.AddOnlyLabels(ts+1, map[string]any{"key3": "value3"})

	errors, err := batch.Write(context.Background())
	if err != nil {
		panic(err)
	}

	if len(errors) != 0 {
		panic("Expected no errors, but got " + string(len(errors)))
	}
}
